core/src/main/scala/kafka/server/ReplicaManager.scala

```diff
@@ -318,6 +318,10 @@ class ReplicaManager(val config: KafkaConfig,
     // A follower can lag behind leader for up to config.replicaLagTimeMaxMs x 1.5 before it is removed from ISR
     scheduler.schedule("isr-expiration", maybeShrinkIsr _, period = config.replicaLagTimeMaxMs / 2, unit = TimeUnit.MILLISECONDS)
     scheduler.schedule("isr-change-propagation", maybePropagateIsrChanges _, period = 2500L, unit = TimeUnit.MILLISECONDS)
+
+    // If IBP < 1.0, controller will send LeaderAndIsrRequest V0 which does not include isNew field.
+    // In this case broker can not determine whether it is safe to create partition when there is log directory failure.
+    // Thus we choose to have broker halt on log diretory failure if IBP < 1.0
```

Comment:
A suggestion (assuming that it's still correct):

"If the inter-broker protocol (IBP) version is less than 1.0, the controller will send a LeaderAndIsrRequest V0 which does not include the `isNew` field. In this case, the broker receiving the request cannot determine whether it is safe to create a partition if a log directory has failed. Thus, we choose to halt the broker on any log directory failure if IBP < 1.0."

Reply:
Thanks much for the detailed suggestion! The new comment is certainly better. I will update the patch as suggested. Will be more careful with the comment going forward.